mobile/native/src/ln_dlc/mod.rs

```diff
@@ -442,5 +449,36 @@ pub async fn trade(trade_params: TradeParams) -> Result<(), (FailureReason, anyh
 
     tracing::info!("Sent trade request to coordinator successfully");
 
+    tracing::info!("Paying order-matching fee");
+
+    let order_matching_fee_invoice = response.text().await.map_err(|e| {
+        (
+            FailureReason::TradeResponse,
+            anyhow!("Could not deserialize order-matching fee invoice: {e:#}"),
+        )
+    })?;
+    let order_matching_fee_invoice = order_matching_fee_invoice.parse().map_err(|e| {
+        (
+            FailureReason::TradeResponse,
+            anyhow!("Could not parse order-matching fee invoice: {e:#}"),
+        )
+    })?;
+
+    NODE.get()
```

Comment:
🙃 I think it would be better to charge this only once the `Order` was filled!

I think we should easily be able to add / trigger the payment from `trade::order::handler::order:filled` instead of triggering it right after the trade.

The problem I see here is, that the fee payment may succeed, but the execution then may fail because of e.g. insufficient balance. I think it's better to charge once we are sure that the execution succeeded (i.e. upon finalize, that's when we trigger `order_filled`.

Reply:
This assumes that we are able to send payments while a position is open - I assume this works.

Reply:
Yep, I agree. I tried to capture this in the TODO in the PR description:

> Only claim the order-matching fee once the trade has been executed.

I will do so before merging this.

Reply:
> Actually, in the history above the fee is displayed after the position was opened - how is that possible? - am I misunderstanding when the payment is triggered? 

Good catch. I suppose the payment may be sent now, but it may only be claimed a bit later. So in theory you could end up resolving the trade before the fee payment.

Reply:
This is all fixed with https://github.com/get10101/10101/pull/863/commits/bb2e157e8abaca2a599764a0c1ea6222b59e01c5, including the visual bug.